infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServerCache.java

```diff
@@ -32,11 +34,12 @@
     //Empirical threshold of number of streams in a logdata beyond which server performance may be slow
     private final int MAX_STREAM_THRESHOLD = 20;
 
-    public LogUnitServerCache(LogUnitServerConfig config, StreamLog streamLog) {
+    public LogUnitServerCache(LogUnitServerConfig config, StreamLog streamLog, Executor executor) {
         this.streamLog = streamLog;
         this.dataCache = Caffeine.newBuilder()
                 .<Long, ILogData>weigher((addr, logData) -> getLogDataTotalSize(logData))
                 .maximumWeight(config.getMaxCacheSize())
+                .executor(executor)
```

Comment:
@ben-manes can a self-Induced deadlock happen if the writers and eviction threads use the same executor?

Reply:
That should be fine. Some turn off the executor and use `Runnable::run` so that the work is amortized on the callers. That's fast on our side, but penalizes the caller when invoking your removal listener, etc. We defer to the common threadpool because it is available and lets us avoid penalizing the caller if your callbacks are expensive, but it isn't necessary nor should you observe deadocks based on how it is configured.

Reply:
K, thanks.